fronted.go

```diff
@@ -386,12 +390,31 @@ func (f *fronted) doDial(m MasqueradeInterface) (conn net.Conn, retriable bool,
 			log.Debugf("Unexpected error dialing, keeping masquerade: %v", err)
 			retriable = true
 		}
-	} else {
-		log.Debugf("Got successful connection to: %v", m)
+		return
 	}
+	log.Debugf("Got successful connection to: %+v", m)
 	return
 }
 
+func isNetworkUnreachable(err error) bool {
+	var opErr *net.OpError
+	if errors.As(err, &opErr) {
+		if errors.Is(opErr.Err, syscall.ENETUNREACH) || errors.Is(opErr.Err, syscall.EHOSTUNREACH) {
+			return true
+		}
+
+		// Fallback to message-based checking (works for Windows and Unix-like systems)
+		errMsg := opErr.Err.Error()
+		if strings.Contains(errMsg, "network is unreachable") ||
+			strings.Contains(errMsg, "no route to host") ||
+			strings.Contains(errMsg, "unreachable network") ||
+			strings.Contains(errMsg, "unreachable host") {
+			return true
+		}
```

Comment:
Which OSes would this not work for? MacOS, iOS, and Android are all Unix based. AFAIK, this _should_ work with all platforms we support.

Reply:
Those string validation should cover everything. I guess the errors check above cover Linux/Unix and this string errors should cover the windows

Reply:
Looking at similar errors at honeycomb you should be able to see errors grouped by OS

Reply:
> I guess the errors check above cover Linux/Unix and this string errors should cover the windows
Oh OK. Would you add that to the comment to make it more clear?